lib/streamlit/DeltaGenerator.py

```diff
@@ -2366,11 +2366,14 @@ def number_input(
 
             if format in ["%d", "%u", "%i"] and float_value:
                 # Fix for https://github.com/streamlit/streamlit/issues/930
-                # If user submitted format is int, assume the intention is to
-                # coerce submitted value to int.
-                value = int(value)
-                int_value = True
-                float_value = False
+                # Warn user to check if displaying float as int was really intended.
+                import streamlit as st
+
+                st.warning(
```

Comment:
Is this a forever warning, so no way to turn off?

Reply:
@jrhone Yeah, it is.  If it's annoying I'm sure we'll hear about it.  But I think it's only going to come up for people who have done something by accident and will appreciate the warning.  At least, that's the bet we're making for now.  :)